src/utils/resin.ts

```diff
@@ -0,0 +1,28 @@
+const RESIN_PREFIX = 'data-resin-';
+
+export function getResinAttribute(name: string): string {
+    return `${RESIN_PREFIX}${name}`;
+}
+
+export function stringify(value: unknown): string | null {
+    if (typeof value === 'object' || typeof value === 'function') {
+        return null;
+    }
+
+    return typeof value === 'string' ? value : String(value);
+}
+
+export function applyResinTags(element: HTMLElement, attributes: Record<string, unknown>): void {
```

Comment:
The type of resin attributes can only be `string | number`, right?

Reply:
Perhaps via the API but setting to DOM attributes needs to be a string but that conversion is done below via `stringify`